Oracle Extractor (Continuation of PR #1074)

lbendig opened the pull request:
This PR addresses some changes and fixes to the original PR #1074 .

jinhyukchang commented:
Test was successful with "lbendig:pr_oracle_extr" branch.

lbendig commented:
@jinhyukchang Thanks for testing it. Can you and/or anyone else please check this PR whether it's eligible to merge to the codebase?

jinhyukchang commented:
@lbendig PR looks good to me.
@vasanthrajamani Could you let someone take a look and merge? (Most of changes are from previous PR which already have been reviewed)

jinhyukchang commented:
@vasanthrajamani , Oracle Source has been in PR quite a while. Could you let someone to take a look? Thanks!

ydai1124 commented on gobblin-core/src/main/java/gobblin/source/extractor/extract/jdbc/OracleExtractor.java:
```diff
@@ -0,0 +1,319 @@
+  
+  @Override
+  public Map<String, String> getDataTypeMap() {
+    Map<String, String> dataTypeMap = ImmutableMap.<String, String> builder()
```
Can you make this map as a cache, so it will be created only once.

jinhyukchang commented:
@lbendig Could you address the comment when you have a chance? Thanks!

jinhyukchang commented:
@lbendig Any update?

lbendig commented:
@ydai1124, @jinhyukchang Thanks for the review, I addressed your comments. I'll open a PR for the documentation. BTW, the TeradataWriter also needs to be documented, I'll take care of that.

ydai1124 commented:
@lbendig  Sorry for the late update. Thanks for contributing!

lbendig commented:
@jinhyukchang , @ydai1124 Thank you for the review! Credit also goes to @bjvanov.
